woocommerce/woocommerce
# [NOX] Implementing loading states to the new settings pages

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52955 



### Screenshots

<img width="2173" alt="Screenshot 2024-11-23 at 07 21 05" src="https://github.com/user-attachments/assets/c935057f-15fb-4779-91fa-83be98b0d254">

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure you have enabled the new Reactified settings pages (see the instructions in https://github.com/woocommerce/woocommerce/pull/52824 if you haven't done this previously).
1. Run ` pnpm run --filter=@woocommerce/plugin-woocommerce watch:build`
2. Load the settings page, you should see the loadable state as shown in the screenshot appear before the content loads.
3. Install/Enable a recommended payment gateway (e.g. one of the top 2 choices that appear in the payments gateways list, or one of the options that appear in `Other payment options`. You should see the loadable state briefly appear while we add/remove options from the main payment gateways list.
4. Replace [this block of code](https://github.com/woocommerce/woocommerce/blob/3b891b7f0ddc0557057744c294557727cb6d8060/plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx#L137) as follows:

 ```ts
			{ isFetching ? (
				<ListPlaceholder rows={ 5 } />
			) : (
				<ListPlaceholder rows={ 5 } />
			) }
```

5. Reload the page, the loading state should appear at all times. Verify it matches the designs.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add loading states to the React settings page
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
